backend/api/handlers/user/UpdatePasswordHandler.go

```diff
@@ -1,15 +1,75 @@
 package user
 
 import (
+	"errors"
 	"fmt"
-	"github.com/gin-gonic/gin"
+	"log"
 	"net/http"
+
+	"LiteracyLink.com/backend/api/model"
+	"LiteracyLink.com/backend/db"
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
 )
 
 func UpdatePasswordHandler(c *gin.Context) {
-	user_id := c.Param("user_id")
-	c.JSON(http.StatusAccepted, gin.H{
-		"status":  "success",
-		"message": fmt.Sprintf("TODO: make func let user with id: %s update password", user_id),
-	})
+	
+	var request model.PassUpdate
+	err := c.BindJSON(&request)
+	if err != nil {
+		log.Default()
+		log.Println(c.Request.Body)
+		c.JSON(http.StatusBadRequest, gin.H{
+			"status":  "error",
+			"message": fmt.Sprintf("Error while binding password update request: %s", err),
+		})
+		return
+	}
+	
+	//get the id (therby jwt token)
+	var id = c.MustGet("userID").(string);
+
+	//set it into the passupdate struct
+	request.UserID = id;
+
+	// Validate password
+	if !isValidPassword(request.PasswordHash) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid password"})
+		return
+	}
+
+	// Hash the password
+	hashedPassword, err := hashPassword(request.PasswordHash)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Error hashing password"})
+		return
+	}
+
+	// Set the hashed password in the User struct
+	request.PasswordHash = hashedPassword
+
+	// Now you can use the request to update the password
+	dbc := c.MustGet("db").(*gorm.DB)
+	err = db.UpdatePass(request, dbc)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"status":  "failed",
+				"message": fmt.Sprintf("incorrect username"),
```

Comment:
error message should probably be something like unable to find user with ID: [IDHERE]